### PR TITLE
Index out of bounds exception if a negative overlay origin is used

### LINKED ISSUE #1559: Index out of bounds exception if a negative overlay origin is used
**Describe the bug**
When a DICOM file contains overlay data with a negative origin, rendering crashes with an index out of bounds exception

**To Reproduce**
See unit test.

**Expected behavior**
Only pixel values that are inside the bounds of the images should be written.

**Screenshots or test DICOM files**
See unit test

**Environment**
Fellow Oak DICOM version: latest development
OS: Any
Platform: Any

### PR DESCRIPTION
**Pull request by amoerie**
Fixes #1559

#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
- [x] I have updated API documentation
- [x] I have included unit tests
- [x] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file

#### Changes proposed in this pull request:
- Check if overlay origin does not cause a negative offset when rendering overlays

### CODE REVIEW

File: FO-DICOM.Core/Imaging/Render/OverlayGraphic.cs [hunk 1]
```diff
@@ -86,7 +86,11 @@ public void Render(int[] pixels, int width, int height)
                     if (data[i] > 0)
                     {
                         if (ox + x >= width) break;
-                        pixels[p] |= _color;
+
+                        if (p >= 0)
```
Comment (mrbean-bremen) [inline FO-DICOM.Core/Imaging/Render/OverlayGraphic.cs:90]: I would actually have expected the check with ox and oy, but this is also ok.

**Review by mrbean-bremen — APPROVED** [review]
Good catch - I remember running into a similar issue long ago (not with fo-dicom, of course).

File: FO-DICOM.Core/Imaging/Render/OverlayGraphic.cs (hunk 1 above)
Comment (amoerie) [reply FO-DICOM.Core/Imaging/Render/OverlayGraphic.cs:90]: If you feel strongly about it I can still change it. :-)

Comment (mrbean-bremen) [reply FO-DICOM.Core/Imaging/Render/OverlayGraphic.cs:90]: I don't :)

**Review by gofal — APPROVED** [review]

**Pull request merged**